### PR TITLE
#7211: Drop overlapping URL patterns

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

Mentioned in the previous PR, this PR cleans up some patterns left behind during the `all_urls` change. See review for details.

If left unsolved, it could cause trouble with:

- https://github.com/pixiebrix/pixiebrix-extension/pull/6851

## Checklist

- [x] Designate a primary reviewer: @twschiller

### CODE REVIEW

File: src/manifest.json [hunk 1]
```diff
@@ -47,7 +46,6 @@
     "tabs",
     "webNavigation",
     "contextMenus",
-    "*://*/*",
     "<all_urls>"
```
Comment (fregante) [inline src/manifest.json:49]: `<all_urls>` is a superset of `*://*/*`

File: src/testUtils/factories/modDefinitionFactories.ts [hunk 2]
```diff
@@ -170,7 +170,7 @@ export const innerStarterBrickModDefinitionFactory = ({
         definition: {
           type: "menuItem",
           isAvailable: {
-            matchPatterns: ["https://*/*"],
+            matchPatterns: ["*://*/*"],
```
Comment (fregante) [inline src/testUtils/factories/modDefinitionFactories.ts:173]: Intentionally https-only?

File: scripts/__snapshots__/manifest.test.js.snap [hunk 3]
```diff
@@ -174,20 +169,16 @@ exports[`customizeManifest mv3 1`] = `
       ],
       "match_about_blank": true,
       "matches": [
-        "https://*/*",
-        "http://*/*",
```
Comment (fregante) [inline scripts/__snapshots__/manifest.test.js.snap:178]: The MV3 is automatically upgraded via `webext-permissions`

- https://github.com/pixiebrix/pixiebrix-extension/pull/7211#discussion_r1437146401

File: src/manifest.json [hunk 4]
```diff
@@ -21,7 +21,7 @@
   "content_security_policy": "script-src 'self' 'unsafe-eval'; font-src 'self' https://fonts.gstatic.com; connect-src 'self' http: https:; object-src 'self'; img-src 'self' data: https:; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com; frame-src 'self' https:",
   "content_scripts": [
     {
-      "matches": ["https://*/*", "http://*/*"],
+      "matches": ["*://*/*"],
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/manifest.json:24]: > *://*/* includes both.

It also includes `file`, `data`, and `chrome-extension`. I think we want to keep http/https

File: src/manifest.json [hunk 5]
```diff
@@ -30,9 +30,8 @@
       "run_at": "document_idle"
     },
     {
-      "matches": ["https://*/*"],
+      "matches": ["*://*/*"],
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/manifest.json:33]: Yes, intentionally https

File: src/testUtils/factories/modDefinitionFactories.ts (hunk 2 above)
Comment (twschiller) [reply src/testUtils/factories/modDefinitionFactories.ts:173]: This is a test file with example data. Let's leave it as https

Comment (fregante) [reply src/testUtils/factories/modDefinitionFactories.ts:173]: Oh right

File: src/manifest.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (fregante) [reply src/manifest.json:24]: > It also includes `file`, `data`, and `chrome-extension`

You're confusing them with `all_urls` I think, refer to the link


> unless there's a compelling reason to change it?

Consistency across the extension maybe

[the remark this answers is not in the document]
Comment (fregante) [reply src/manifest.json:24]: I double-checked the `tabs.query()` call in the file below too. All good:

Screenshot 7

[the remark this answers is not in the document]
Comment (twschiller) [reply src/manifest.json:24]: > That's all_urls, refer to the link

Gotcha - misread that link. Approving

**Review by twschiller — APPROVED** [review]

**Pull request merged**